src/eureka/S4cal_StellarSpectra/s4cal_StellarSpec.py

```diff
@@ -0,0 +1,285 @@
+                                 spec.optspec.time_units,
+                                 name='Std. Dev. of In-Eclipse Flux')
+
+        # Create XArray dataset
+        ds = dict(base_flux=base_flux, base_fstd=base_fstd,
+                  ecl_flux=ecl_flux, ecl_fstd=ecl_fstd)
+        ds = xrio.makeDataset(ds)
+        batch.append(ds)
+
+    # Concatenate datasets
+    ds = xrio.concat(batch, dim="time")
+    filename = meta.outputdir+'S4cal_'+meta.eventlabel + "_CalStellarSpec.h5"
+    xrio.writeXR(filename, ds, verbose=True)
+
+    if meta.isplots_S4cal >= 2:
+        plot_stellarSpec(meta, ds)
+
+    return meta, spec, ds
+
+
+def plot_whitelc(optspec, time, meta, i, fig=None, ax=None):
+    '''Plot binned white light curve and indicate
+    baseline and in-eclipse regions.
+    '''
+    toffset = meta.time_offset
+    it0, it1, it2, it3, it4, it5 = meta.it
+
+    # Created binned white LC
+    lc = np.ma.sum(optspec, axis=1)
+    lc /= np.mean(lc)
+    lc_bin = util.binData_time(lc, time, nbin=meta.nbin_plot)
+    time_bin = util.binData_time(time, time, nbin=meta.nbin_plot)
+
+    if i == 0:
+        fig = plt.figure(4202, figsize=(8, 5))
+        plt.clf()
+        ax = fig.subplots(1, 1)
+        ax.plot(time_bin-toffset, lc_bin, '.', color='0.2', alpha=0.8,
+                label='Binned White LC')
+    ymin, ymax = ax.get_ylim()
+    ax.fill_betweenx((ymin, ymax), time[it0]-toffset, time[it1]-toffset,
+                     color=colors[1], alpha=0.2)
+    ax.fill_betweenx((ymin, ymax), time[it4]-toffset, time[it5]-toffset,
+                     color=colors[1], alpha=0.2)
+    ax.fill_betweenx((ymin, ymax), time[it2]-toffset, time[it3]-toffset,
+                     color=colors[0], alpha=0.2)
+    ax.vlines([time[it1]-toffset, time[it4]-toffset,
+              time[it0]-toffset, time[it5]-toffset],
+              ymin, ymax, color=colors[1], label='Baseline Regions')
+    ax.vlines([time[it2]-toffset, time[it3]-toffset],
+              ymin, ymax, color=colors[0], label='In-Eclipse Region')
+    if i == 0:
+        ax.set_ylim(ymin, ymax)
+        ax.legend(loc='best')
+        ax.set_xlabel("Time (MJD)")
+        ax.set_ylabel("Normalized Flux")
+    fname = 'figs'+os.sep+'fig4202_WhiteLC'
+    fig.savefig(meta.outputdir+fname+plots.figure_filetype,
+                bbox_inches='tight', dpi=300)
+    return fig, ax
+
+
+def plot_stellarSpec(meta, ds):
+    '''Plot calibrated stellar spectra from
+    baseline and in-eclipse regions.
+    '''
+    fig = plt.figure(4201, figsize=(8, 5))
+    plt.clf()
+    ax = fig.subplots(1, 1)
+    for i in range(len(ds.time)):
+        ax.errorbar(ds.wavelength, ds.base_flux[:, i], ds.base_fstd[:, i],
+                    fmt='.', ms=1, label=f'Baseline ({ds.time.values[i]})',
+                    color=colors[1])
+        ax.errorbar(ds.wavelength, ds.ecl_flux[:, i], ds.ecl_fstd[:, i],
+                    fmt='.', ms=1, label=f'In-Eclipse ({ds.time.values[i]})',
```

Comment:
Would "In-Occultation" be better, given that this may be applied to transit data as well?